openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -392,6 +394,17 @@ def validate_user_preference_serializer(serializer, preference_key, preference_v
                 "user_message": user_message,
             }
         })
+    if preference_key == "time_zone" and preference_value not in common_timezones_set:
+        developer_message = ugettext_noop(u"Value '{preference_value}' not valid for preference '{preference_key}': Not in timezone set.")  # pylint: disable=line-too-long
+        user_message = ugettext_noop(u"Value '{preference_value}' is not valid for user preference '{preference_key}'.")
```

Comment:
@kevinjkim @jcdyer FYI - if the doc is still being decided, feel free to create a separate PR to update the Doc.  I'd like to close out this long-term branch and make further changes in subsequent smaller PRs, especially since it's behind a feature flag.